app/AvailablePlugin/PasswordAuthenticator/Model/PasswordResetToken.php

```diff
@@ -205,12 +205,19 @@ public function generateRequest($authenticatorId, $q) {
       throw new InvalidArgumentException(_txt('er.notfound', array(_txt('ct.authenticators.1'), $authenticatorId)));
     }
     
-    // Next, generate a token in order to send it
-    $token = $this->generate($passwordAuthenticator['PasswordAuthenticator']['id'], $coPersonId);
+    // Get the name of the calling function to determine whether we're sending Reset Token Request or Username Reminder
+    $trace = debug_backtrace();
```

Comment:
Instead of using debug_backtrace, add a parameter to the function signature indicating how it's supposed to be called. eg: `public function generateRequest($authenticatorId, $q, $mode='reset')`